hotel/models/hotel_folio.py

```diff
@@ -0,0 +1,810 @@
+        """
+        return self.search_count([("state", "=", "draft")])
+
+    @api.model
+    def _get_checkin_date(self):
+        if self._context.get("tz"):
+            to_zone = self._context.get("tz")
+        else:
+            to_zone = "UTC"
+        return _offset_format_timestamp1(
+            time.strftime("%Y-%m-%d 12:00:00"),
+            DEFAULT_SERVER_DATETIME_FORMAT,
+            DEFAULT_SERVER_DATETIME_FORMAT,
+            ignore_unparsable_time=True,
+            context={"tz": to_zone},
+        )
+
+    @api.model
+    def _get_checkout_date(self):
+        if self._context.get("tz"):
+            to_zone = self._context.get("tz")
+        else:
+            to_zone = "UTC"
+        tm_delta = timedelta(days=1)
+        return (
+            datetime.strptime(
+                _offset_format_timestamp1(
+                    time.strftime("%Y-%m-%d 12:00:00"),
+                    DEFAULT_SERVER_DATETIME_FORMAT,
+                    DEFAULT_SERVER_DATETIME_FORMAT,
+                    ignore_unparsable_time=True,
+                    context={"tz": to_zone},
+                ),
+                "%Y-%m-%d %H:%M:%S",
+            )
+            + tm_delta
+        )
+
+    name = fields.Char(
+        "Folio Number", readonly=True, index=True, default="New"
+    )
+    order_id = fields.Many2one(
+        "sale.order", "Order", delegate=True, required=True, ondelete="cascade"
+    )
+    checkin_date = fields.Datetime(
+        "Check In",
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        default=_get_checkin_date,
+    )
+    checkout_date = fields.Datetime(
+        "Check Out",
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        default=_get_checkout_date,
+    )
+    room_lines = fields.One2many(
+        "hotel.folio.line",
+        "folio_id",
+        readonly=True,
+        states={"draft": [("readonly", False)], "sent": [("readonly", False)]},
+        help="Hotel room reservation detail.",
+    )
+    service_lines = fields.One2many(
+        "hotel.service.line",
+        "folio_id",
+        readonly=True,
+        states={"draft": [("readonly", False)], "sent": [("readonly", False)]},
+        help="Hotel services details provided to"
+        "Customer and it will included in "
+        "the main Invoice.",
+    )
+    hotel_policy = fields.Selection(
+        [
+            ("prepaid", "On Booking"),
+            ("manual", "On Check In"),
+            ("picking", "On Checkout"),
+        ],
+        "Hotel Policy",
+        default="manual",
+        help="Hotel policy for payment that "
+        "either the guest has to payment at "
+        "booking time or check-in "
+        "check-out time.",
+    )
+    duration = fields.Float(
+        "Duration in Days",
+        help="Number of days which will automatically "
+        "count from the check-in and check-out date. ",
+    )
```

Comment:
Could it be a computed field?

Reply:
As of now we managing this with onchange function.